new_core_concepts/modules/ROOT/pages/typeql/entities-relations-attributes.adoc

```diff
@@ -1,4 +1,137 @@
+
+[,typeql]
+----
+attribute age, value integer;`
+----
+
+== Owning attributes & playing roles
+
+Entity & relation types may be defined to "own" attribute types.
+
+E.g. A `person` may own an attribute `birth-date`, or an `employment` may own a `start-date`;
+[,typeql]
+----
+person owns age;
+----
+
+Entity & relation types may be defined to "play" a role defined by a relation.
+
+E.g. The entity type `company` may play the  `employer` role of an `employment` relation.
+
+[,typeql]
+----
+company plays employment:employer;
+----
+
+== Polymorphism
+
+Polymorphism in TypeDB comes is realised in two ways:
+
+=== Interfaces
+Multiple types may "implement" the same interface. In the following schema snippet, the `employer` role in the `employment` relation can be played by a `company` or an `charity`.
+
+[,typeql]
+----
+define
+  relation employment,
+    relates employer,
+    relates employee;
+
+  entity company sub organisation,
+    plays employment:employer;
+  entity charity sub organisation,
+    plays employment:employer;
+----
+
+Here, the role `employer` is the interface declared by the `employment` type. The `company` and `charity` types implement the interface.
+The `relates` keyword is used to define the new role-type interface.
+The `plays` keyword is used to implement a role-type interface.
+
+[NOTE]
+====
+Attribute types always declare an implicit interface. A type can be defined to implement this interface using the `owns` keyword.
+====
+
+=== Subtyping
+A single-inheritance *subtyping* system allows types to specialize an existing type.
+An instance of the subtype may be used at any interface or function argument that accepts an instance of its super-type.
+
+In the following schema snippet, entities of type `company` or `charity` can play the `employer` in an `employment`, although it is their supertype `organisation` which is defined to play the role.
+
+[,typeql]
+----
+define
+  relation employment,
+    relates employer,
+    relates employee;
+
+  entity organisation sub profile,
+    plays employment:employer;
+  entity company sub organisation;
+  entity charity sub organisation;
+----
+Thus, a subtype must inherit the implementation of any interface its supertype implements.
+
+==== Inheriting & specialising interfaces
+A subtype also inherits any interfaces defined by its supertype.
+It may optionally "specialise" an inherited interface by defining a subtype of that interface.
+
+In the following schema snippet, `content-engagement` inherits the `content` role from `interaction` as-is, and "specialises" the inherited `subject` role into the `author` role.
+
+[,typeql]
+----
+define
+relation interaction @abstract,
+    relates subject,
+    relates content;
+relation content-engagement @abstract, sub interaction,
+    relates author as subject;
+----
+
+[NOTE]
+====
+An attribute type inherits and implicitly specialises the interface of its supertype.
```

Comment:
Not sure it's worth saying as the user never sees this IMO?

Reply:
That's why it's in a note rather than in the main text.
I think it's easier to understand the behaviour of attribute ownerships if they understand how `plays` works and know the mapping to attributes.

Reply:
Fair enough, i think it's a usefu parallel